### ISSUE TITLE
[Bug] Step graph and "connectNulls = false" connects where it should not

### ISSUE THREAD

**Issue by skubidu1324** [apache/echarts#20191]
### Version

5.2.0

### Link to Minimal Reproduction

https://echarts.apache.org/examples/en/editor.html?c=line-step&version=5.5.1

### Steps to Reproduce

Use above example and add null values in the middel.

### Current Behavior

Lines are painted that are not allowed to be painted. I have marked these in red here:

![image](https://github.com/user-attachments/assets/c2c827d2-676f-43fd-be31-668a5a48b73f)

### Expected Behavior

Marked area should not be there.

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

**Comment by helgasoft** [comment]
it's all normal, because **step** option is **different** for each line. Set all lines to same step and the gaps will look the same.
📌 please close issue if problem solved.

**Comment by skubidu1324** [comment]
No, unfortunately that doesn't solve the problem. It is the case that both the connection from the value to the left of null must have no connection, as well as the connection from the value to the right from null to null. This means that the graph is not displayed correctly in any of the step options.

**Comment by Arkadij-Dev-C** [comment]
bump
